cpp/include/kvikio/utils.hpp

```diff
@@ -52,36 +53,84 @@ inline constexpr std::size_t page_size = 4096;
 }
 
 /**
- * @brief Get the current cuda context
+ * @brief Return the device owning the pointer
  *
- * Previously, we got the cuda context from the provided device pointer by calling
- * `cuPointerGetAttribute(..., CU_POINTER_ATTRIBUTE_CONTEXT)`. However, this doesn't
- * work for stream ordered device memory allocations[1] so we now get the current
- * cuda context instead.
- * [1] <https://docs.nvidia.com/cuda/cuda-c-programming-guide/#pointer-attributes>
- *
- * @param check_owning_devPtr If not NULL, a device memory pointer that must have
- * been allocated by, mapped by, or registered with the current context. If this
- * isn't the case, a CUfileException is thrown.
- *
- * @return The current cuda context
+ * @param ptr Device pointer to query
+ * @return The device ordinal
  */
-[[nodiscard]] inline CUcontext get_current_context(const void* check_owning_devPtr = nullptr)
+[[nodiscard]] inline int get_device_ordinal_from_pointer(const void* devPtr)
```

Comment:
Does it make sense to accept a `CUdeviceptr` instead of a `void * const`? The only _current_ caller already has a `CUdeviceptr` hanging around. I guess it's a wash.